services/searchengine/bleveengine/search.go

```diff
@@ -329,6 +329,13 @@ func (b *BleveEngine) SearchChannels(teamId, userID, term string) ([]string, *mo
 	// AND
 	// match (channelType != 'P' || (<> in userIDs && channelType == 'P'))
 
+	// (or if isGuest is true)
```

Comment:
Hopefully my comments were helpful to figure out the logic.

Reply:
Those were top class comments, definitely helped a lot :)